lib/utils/simpleconfig.rb

```diff
@@ -7,6 +7,16 @@
 require 'utils/parser'
 
 class SimpleConfig
+  # for configs that have keys whose values may be hashes
+  # (such as a mysql config that has [sections]), provide
+  # a hash whose values are accessible via methods to provide
+  # a good user experience with `its` blocks.
+  class ConfigHash < Hash
```

Comment:
This is definitely a valid solution. We use [hashie](https://github.com/intridea/hashie#mash) in other areas like https://github.com/chef/inspec/blob/546486ff6afc371103943501e9d7e15bcc6941ff/lib/resources/service.rb#L180-L183. Should we stick to the same solution?